provider/lxd/provider.go

```diff
@@ -165,18 +165,22 @@ func (p *environProvider) Ping(ctx context.ProviderCallContext, endpoint string)
 
 	// Ensure the Port on the Host, if we get an error it is reasonable to
 	// assume that the address in the spec is invalid.
-	var err error
-	endpoint, err = lxd.EnsureHostPort(endpoint)
+	lxdEndpoint, err := lxd.EnsureHostPort(endpoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
+	// Make sure we have an https url
+	if lxdEndpoint != endpoint {
+		return errors.Errorf("invalid URL %q: only HTTPS is supported", endpoint)
```

Comment:
We don't need this error, we shouldn't care about the URL schema, I think we should just fix it.

Reply:
Having thought about this more, @mitechie what are your thoughts about just throwing an error if the url is wrong and that we then don't swap the url behind the scenes, potentially confusing people?